mirgecom/diffusion.py

```diff
@@ -430,6 +431,90 @@ def get_diffusion_flux(
             penalty_amount=penalty_amount)
 
 
+class RobinDiffusionBoundary(DiffusionBoundary):
+    r"""Robin boundary condition for the diffusion operator.
+
+    The non-homogeneous Robin boundary condition is a linear combination of
+    $u$ and its gradient $\nabla u$, given by
+
+    .. math::
+
+        (\alpha u - \kappa \nabla u \cdot \mathbf{\hat{n}})|_\Gamma =
+            \alpha u_{ref}.
+
+    where $u_{ref}$ is the reference value of $u$ for $x \to \infty$ and
+    $\alpha$ is the weight for $u$. The gradient weight $\kappa$ is the
+    conductivity (thermal) or the diffusivity (species).
+
+    The current implementation uses external data
+
+    .. math::
+
+        u^+ = u^-
+
+    when computing the boundary fluxes for $\nabla u$, and
+
+    .. math::
+
+        \nabla u\cdot\mathbf{\hat{n}} |_\Gamma =
+            \frac{\alpha}{\kappa}(u_{ref} - u^-)
+
+    when computing the boundary fluxes for $\nabla \cdot (\kappa \nabla u)$.
+
+    .. automethod:: __init__
+    .. automethod:: get_grad_flux
+    .. automethod:: get_diffusion_flux
+    """
+
+    def __init__(self, value, alpha):
+        """
+        Initialize the boundary condition.
+
+        Parameters
+        ----------
+        value: float or meshmode.dof_array.DOFArray
+            the reference value(s) of $u$ along the boundary
+        alpha: float or meshmode.dof_array.DOFArray
+            the weight for the variable $u$ at the boundary
+        """
+        self.value = value
```

Comment:
I would maybe call this `u_ref`, or something else. `value` is kind of ambiguous for this BC.

Reply:
Changed to `u_ref` as suggested.